api/v1alpha6/conversion_test.go

```diff
@@ -102,6 +110,125 @@ func TestFuzzyConversion(t *testing.T) {
 
 }
 
+func NnfContainerProfileFuzzFunc(_ runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		NnfContainerProfilev6Fuzzer,
+	}
+}
+
+func NnfContainerProfilev6Fuzzer(in *NnfContainerProfileData, c fuzz.Continue) {
+	// Tell the fuzzer to begin by fuzzing everything in the object.
+	c.FuzzNoCustom(in)
+
+	noFuzzContainer := func(containers []corev1.Container) {
+		for i := range containers {
+			// We only use the following fields in v1alpha7 and later, the rest are set to nil or default values:
+			// - Name
+			// - Image
+			// - Command
+			// - Args
+			// - Env
+			// - EnvFrom
+			// - VolumeMounts
+			containers[i].Resources = corev1.ResourceRequirements{}
+			containers[i].RestartPolicy = nil
+			containers[i].ResizePolicy = nil
+			containers[i].VolumeDevices = nil
+			containers[i].ReadinessProbe = nil
+			containers[i].StartupProbe = nil
+			containers[i].LivenessProbe = nil
+			containers[i].Lifecycle = nil
+			containers[i].TerminationMessagePath = ""
+			containers[i].TerminationMessagePolicy = ""
+			containers[i].ImagePullPolicy = ""
```

Comment:
Some of these we want, maybe in the next PR, definitely in the same release.
This one is already documented here: https://nearnodeflash.github.io/latest/guides/user-containers/readme/#using-a-private-container-repository

Reply:
Looks like `imagePullSecrets` for sure?

Reply:
We could support the probe fields (3, I count), and the TerminationMessage* fields.
I don't mind doing that in the next PR.

Reply:
Yea, let's save that for a follow up PR

Reply:
I added an issue for this: https://github.com/NearNodeFlash/NearNodeFlash.github.io/issues/260